polynote-frontend/polynote/messaging/receiver.ts

```diff
@@ -646,6 +646,12 @@ export class ServerMessageReceiver extends MessageReceiver<ServerState> {
             return { serverOpenNotebooks: notebooks }
         })
         this.receive(messages.SearchNotebooks, (s, query, notebookSearchResults) => {
+            // if search returns successfully, then remove ParsingFailures
+            ErrorStateHandler.get.state.serverErrors.forEach(serverError => {
```

Comment:
Hmm, I wonder whether this can be done in `submitSearch`. 

If the search modal kept track of the errors it displayed, it could then remove them from the `ErrorStateHandler` directly upon submission of a new search (or, receipt of search results if you'd prefer that UX, right next to the `clearAllRows` call?)... and then all the logic could be consolidated in one place. What do you think?

Reply:
Ah!! That makes a lot of sense to me :))